src/win32crypto/cert.c

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) 2025. NetFoundry Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//         https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#include <tlsuv/tls_engine.h>
+
+#include "cert.h"
+#include "../alloc.h"
+#include "../um_debug.h"
+
+#include <time.h>
+#include <bcrypt.h>
+#include <ncrypt.h>
+
+static void free_cert(tlsuv_certificate_t cert) {
+    win32_cert_t *c = (win32_cert_t *) cert;
+    if (c->store) {
+        CertCloseStore(c->store, CERT_CLOSE_STORE_FORCE_FLAG);
+    }
+    tlsuv__free(c);
+}
+
+static int cert_to_pem(const struct tlsuv_certificate_s *cert, int full, char **pem, size_t *pemlen) {
+    win32_cert_t *c = (win32_cert_t *) cert;
+    HCERTSTORE store = c->store;
+    unsigned int flags = CRYPT_STRING_BASE64HEADER | CRYPT_STRING_NOCR;
+
+    PCCERT_CONTEXT cert_ctx = CertEnumCertificatesInStore(store, NULL);
```

Comment:
why are you not calling CertCloseStore? Shouldn't you be?

Reply:
no, why?

Reply:
Ah, I confused it with CertOpenStore. Gotcha thx